### PR TITLE
feat(api): get certslug route

### LINKED ISSUE #50501: GET /certificate/showCert/:username/:certSlug

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #50501

### CODE REVIEW

File: api/src/utils/error-formatting.ts [hunk 1]
```diff
@@ -47,6 +50,30 @@ export const formatProjectCompletedValidation = (
...
+ * @param errors An array of validation errors.
+ * @returns Formatted errors that can be used in the response.
+ */
+export const formatCertificationValidation = (
+  errors: ErrorObject[]
+): FormattedError => {
+  const error = getError(errors);
+
+  return error.instancePath === '' &&
+    Object.hasOwn(certTypes, error.params.missingProperty as CertLogs)
```
Comment (Sboonny) [inline api/src/utils/error-formatting.ts:65]: hasOwn checks the `properties`, I need to use [valus](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values)

File: api/src/schemas.ts [hunk 2]
```diff
@@ -625,5 +628,46 @@ export const schemas = {
...
+        Type.Literal('information-security-v7'),
+        Type.Literal('machine-learning-with-python-v7'),
+        Type.Literal('college-algebra-with-python-v8'),
+        // Type.Literal('foundational-c-sharp-with-microsoft'),
+        Type.Literal('legacy-back-end'),
+        Type.Literal('legacy-data-visualization'),
+        Type.Literal('information-security-and-quality-assurance'),
+        Type.Literal('full-stack')
+      ]),
+      username: Type.String()
```
Comment (Bushbaby02A) [inline api/src/schemas.ts:656]: `string`

**Review by Starboiyonyesom — COMMENTED** [review]
It's a fair plan and am definitely sure am gonna enjoy it

File: api/src/app.ts [hunk 3]
```diff
@@ -202,6 +202,7 @@ export const build = async (
   void fastify.register(settingRoutes);
   void fastify.register(donateRoutes);
   void fastify.register(userRoutes);
+  void fastify.register(certificateRoutes);
```
Comment (ojeytonwilliams) [inline api/src/app.ts:205]:
```suggestion
```
This is already registered on line 200

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by Sembauke — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**